httpcore5/src/main/java/org/apache/hc/core5/net/URLEncodedUtils.java

```diff
@@ -132,46 +126,12 @@
                 }
             }
             if (!name.isEmpty()) {
-                list.add(new BasicNameValuePair(
-                        decodeFormFields(name, charset),
-                        decodeFormFields(value, charset)));
-            }
-        }
-        return list;
-    }
-
-    static List<String> splitSegments(final CharSequence s, final BitSet separators) {
-        final Tokenizer.Cursor cursor = new Tokenizer.Cursor(0, s.length());
-        // Skip leading separator
-        if (cursor.atEnd()) {
-            return Collections.emptyList();
-        }
-        if (separators.get(s.charAt(cursor.getPos()))) {
-            cursor.updatePos(cursor.getPos() + 1);
-        }
-        final List<String> list = new ArrayList<>();
-        final StringBuilder buf = new StringBuilder();
-        for (;;) {
-            if (cursor.atEnd()) {
-                list.add(buf.toString());
-                break;
-            }
-            final char current = s.charAt(cursor.getPos());
-            if (separators.get(current)) {
-                list.add(buf.toString());
-                buf.setLength(0);
-            } else {
-                buf.append(current);
+                list.add(new BasicNameValuePair(PercentCodec.decode(name, charset), PercentCodec.decode(value, charset)));
```

Comment:
Confusing, the previous code refers to forms, but the new one does not use `WWWCodec`. Is that expected?

Reply:
@michael-o I am sorry I am not sure I understand the problem. The entire class has been deprecated and the code you are referring to is left there for backward compatibility.

Reply:
Yes, that's the point. Form encoding does not fully correspond to URI encodig (query string). I would expect a space to be converted to plus for forms, like previously, but `PercentCodec` won't do that. You have replaced `decodeFormFields()` with `PercentCodec`, this is not fully equivalient for me. I am looking https://url.spec.whatwg.org/#string-percent-encode-after-encoding (`spaceAsPlus`). Am I wrong?

Reply:
@michael-o My intention was to support `spaceAsPlus`for input but use standard percent encoding for output, but I will revert it.

Reply:
I don't consider this necessary to be wrong, I wanted to clarify a simple misundestanding. As you can see by the spec, who passes `spaceAsPlus` is not specified. As fas as I can see form the spec, UTF-8 shall be default and `spaceAsPlus` is optional which means `false`. I consider you did the right thing. But we need to resonsider the default encoding of that content type.

Reply:
@michael-o Oh, I made a mistake. `spaceAsPlus` should be `true` for backward compatibility.